File: src/solar/util/game/DestructableUtil.ts
import RectUtil from "./RectUtil";
import {Rectangle} from "solar/w3ts/handles/rect";

/** @noSelf **/
export default class DestructableUtil {


    /**
     * 在指定区域是否有装饰物
     * @param centerX
     * @param centerY
     * @param radius
     * @param destructableId
     */
    static hasDestructableInRect(centerX: number, centerY: number, radius: number, destructableId: string): boolean {
        let r: Rectangle = RectUtil.GetRectFromCircle(centerX, centerY, radius)
        tempBoolean = false;
        tempNumber = FourCC(destructableId);
        //使用命名函数
        r.enumDestructables(null, DestructableUtil._HasDestructableInRectActionFunc)
        r.destroy()
        return tempBoolean
    }

    private static _HasDestructableInRectActionFunc(this: void) {
        if (GetDestructableTypeId(GetEnumDestructable()) == tempNumber) {
            tempBoolean = true
        }
    }

}

File: scripts/solar/util/game/DestructableUtil.lua
local ____lualib = require("lualib_bundle")
local __TS__Class = ____lualib.__TS__Class
local __TS__SourceMapTraceBack = ____lualib.__TS__SourceMapTraceBack
__TS__SourceMapTraceBack(debug.getinfo(1).short_src, {["6"] = 1,["7"] = 1,["8"] = 5,["9"] = 5,["10"] = 5,["12"] = 5,["13"] = 15,["14"] = 16,["15"] = 17,["16"] = 18,["17"] = 20,["18"] = 21,["19"] = 22,["20"] = 15,["21"] = 25,["22"] = 26,["23"] = 27,["25"] = 25,["26"] = 5});
local ____exports = {}
local ____RectUtil = require("solar.util.game.RectUtil")
local RectUtil = ____RectUtil.default
____exports.default = __TS__Class()
local DestructableUtil = ____exports.default
DestructableUtil.name = "DestructableUtil"
function DestructableUtil.prototype.____constructor(self)
end
function DestructableUtil.hasDestructableInRect(centerX, centerY, radius, destructableId)
    local r = RectUtil.GetRectFromCircle(centerX, centerY, radius)
    tempBoolean = false
    tempNumber = FourCC(destructableId)
    r:enumDestructables(nil, ____exports.default._HasDestructableInRectActionFunc)
    r:destroy()
    return tempBoolean
end
function DestructableUtil._HasDestructableInRectActionFunc()
    if GetDestructableTypeId(GetEnumDestructable()) == tempNumber then
        tempBoolean = true
    end
end
____exports.default = DestructableUtil
return ____exports
